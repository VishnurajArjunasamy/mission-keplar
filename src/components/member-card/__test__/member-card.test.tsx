import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import MemberCard from "../member-card";

const mockStore = configureStore([]);

describe("Member Card Component", () => {
  let store: ReturnType<typeof mockStore>;

  beforeEach(() => {
    store = mockStore({
      sideBar: {
        isDarkMode: false,
      },
    });
  });

  it("renders member card with image, name and city", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemberCard image="smaple-img" name="user1" city="chennai" />
      </Provider>
    );
    expect(screen.getByAltText("user1")).toBeInTheDocument();
    expect(screen.getByText("user1")).toBeInTheDocument();
    expect(screen.getByText("chennai")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders member card with light mode styles", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <MemberCard image="smaple-img" name="user1" city="chennai" />
      </Provider>
    );
    const memberCard = screen.getByText("user1").parentElement;
    expect(memberCard).toHaveClass("light");
    expect(memberCard).not.toHaveClass("dark");

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders member card with dark mode styles", () => {
    store = mockStore({
      sideBar: {
        isDarkMode: true,
      },
    });
    const { asFragment } = render(
      <Provider store={store}>
        <MemberCard image="smaple-img" name="user1" city="chennai" />
      </Provider>
    );
    const memberCard = screen.getByText("user1").parentElement;
    expect(memberCard).not.toHaveClass("light");
    expect(memberCard).toHaveClass("dark");
    expect(asFragment()).toMatchSnapshot();
  });
});
