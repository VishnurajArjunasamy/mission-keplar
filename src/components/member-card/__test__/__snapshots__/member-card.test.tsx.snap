// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Card Component renders member card with dark mode styles 1`] = `
<DocumentFragment>
  <div
    class="member dark"
  >
    <img
      alt="user1"
      class="img"
      src="smaple-img"
    />
    <h1>
      user1
    </h1>
    <h2>
      chennai
    </h2>
  </div>
</DocumentFragment>
`;

exports[`Member Card Component renders member card with image, name and city 1`] = `
<DocumentFragment>
  <div
    class="member light"
  >
    <img
      alt="user1"
      class="img"
      src="smaple-img"
    />
    <h1>
      user1
    </h1>
    <h2>
      chennai
    </h2>
  </div>
</DocumentFragment>
`;

exports[`Member Card Component renders member card with light mode styles 1`] = `
<DocumentFragment>
  <div
    class="member light"
  >
    <img
      alt="user1"
      class="img"
      src="smaple-img"
    />
    <h1>
      user1
    </h1>
    <h2>
      chennai
    </h2>
  </div>
</DocumentFragment>
`;
